lib/bank_accounts/checking_account.rb

```diff
@@ -0,0 +1,50 @@
+module BankAccounts
+
+	class CheckingAccount < BankAccounts::Account
+		attr_reader :free_checks_remaining, :using_check
+
+		def initialize(id, initial_balance)
+			super
+			@minimum 				= 0
+			@overdraft_limit	 	= -10
+			@fee					= 1
+			@check_fee				= 2
+			@free_checks_remaining 	= 3
+			@using_check 			= false
+		end
+
+		def validate_withdrawal(amount)
+			if @using_check
+				return @balance if amount == 0
+				valid = true
+				amount_to_withdraw = @free_checks_remaining > 0 ? amount : amount + @check_fee
+				if amount_to_withdraw <= @balance - @overdraft_limit
+					return amount_to_withdraw, valid
+				else
+					puts "You cannot withdraw more than the limit: #{@overdraft_limit}."
+					valid = false
+					return @balance, valid
+				end
+			else
+				super
+			end
+		end
+
+		def withdraw_using_check(amount)
+			@using_check = true
```

Comment:
Since withdraws using checks are only done in checking accounts, the branch for `if @using_check` in validate_withdrawal could just live in this function and you could use the inherited validate_withdrawal.

This also helps avoid some pitfalls with `raise`: if validate_withdrawal or a deeper function raises an error, `@using_check` does not get reset to false on line 43.